ovs/extensions/storageserver/storagedriverconfig/storagedriverconfig.py

```diff
@@ -0,0 +1,336 @@
+        :param serialize_read_cache: Whether to serialize the readcache on exit or not
+        :param clustercache_mount_points: An array of directories and sizes to be used as Read Cache mount points
+        :param kwargs:
+        """
+        self.read_cache_serialization_path = read_cache_serialization_path
+        self.serialize_read_cache = serialize_read_cache
+        self.clustercache_mount_points = clustercache_mount_points
+
+
+class StorageDriverConfig(object):
+    def __init__(self, local_connection_path, vrouter_cluster_id,
+                 backend_config, vrouter_config, volume_manager_config, filesystem_config, dtl_config, backend_interface_config, mds_config, dls_config,
+                 events_config, network_config, vregistry_config, filedriver_config, contentcache_config, scrub_manager_config,
+                 *args, **kwargs):
+
+        self.local_connection_path = local_connection_path  #enkel voor local backend
+        self.vrouter_cluster_id = vrouter_cluster_id
+
+        # Optional params
+
+        self.backend_type = kwargs.get('backend_type', None)
+        self.num_threads = kwargs.get('num_threads', None)
+        self.shm_region_size = kwargs.get('shm_region_size', None)
+
+        self.fuse_min_workers = kwargs.get('fuse_min_workers', None)
+        self.fuse_max_workers = kwargs.get('fuse_max_workers', None)
+
+        self.stats_collector_destination = kwargs.get('stats_collector_destination', None)
+        self.stats_collector_interval_secs = kwargs.get('stats_collector_interval_secs', None)
+
+        self.asio_service_manager_threads = kwargs.get('asio_service_manager_threads', None)
+        self.asio_service_manager_io_service_per_thread = kwargs.get('asio_service_manager_io_service_per_thread', None)
+
+        self.bgc_threads = kwargs.get('bgc_threads')
+
+        # Seperate configs
+
+        self.vrouter_config = vrouter_config
+        self.backend_config = backend_config
+        self.volume_manager_config = volume_manager_config
+        self.filesystem_config = filesystem_config
+        self.dtl_config = dtl_config
+        self.backend_interface_config = backend_interface_config
+        self.mds_config = mds_config
+        self.dls_config = dls_config
+        self.events_config = events_config
+        self.network_config = network_config
+        self.vregistry_config = vregistry_config
+        self.filedriver_config = filedriver_config
+        self.contentcache_config = contentcache_config
+        self.scrub_manager_config = scrub_manager_config
+        self._config = {}
+
+        self._config = {'asio_service_manager': {'asio_service_manager_io_service_per_thread': self.asio_service_manager_io_service_per_thread,
+                                                 'asio_service_manager_threads': self.asio_service_manager_threads},
+                        'backend_connection_manager': {'0': #todo what to do with the proxies?
```

Comment:
Add a method to add a proxy config to the backend_connection_manager

Reply:
The backend_connection_manager should be entirely generated by self.backend_connection_manager.get_config()